src/ledmon/ledmon.c

```diff
@@ -753,8 +755,45 @@ static void _send_msg(struct block_device *block)
 				  host ? host : block->sysfs_path);
 		}
 	}
-	block->send_fn(block, block->ibpi);
+
+	status = block->send_fn(block, block->ibpi);
+	if (status == STATUS_INVALID_STATE) {
+		switch (block->ibpi) {
+		/* Following two patterns generally means nothing to do with status LED device. */
```

Comment:
It doesn't clarify why it has to be this way. To avoid error we would skip executing function for those states. We care the most about ibpi_prev which is updated on success and we have to highlight that we don't want regression.

```c
/**
 * Majority of controllers return success for those special states and ibpi_prev is
 * updated, hence here we translate error to success and force ibpi_prev update for every controller.
 * This is prevention from controller dependent issues.
 */
```

Reply:
I analyzed it again, and did it so as not to print log_error at line 800 for these patterns and at the same time leave the current approach to updating ibpi_prev (update regardless of send_message_fn).

Update ibpi_prev is not preceded by a any status verification.